testnet-contracts/contracts/Oracle.sol

```diff
@@ -0,0 +1,297 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../../node_modules/openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./Recoverer.sol";
+
+contract Oracle is Recoverer {
```

Comment:
Should Recoverer be a library rather than something we inherit?

Reply:
Yes, see comment above.